src/ui/states/state-present-offer.svelte

```diff
@@ -1,51 +1,81 @@
 <script lang="ts">
-  import ModalSection from "../modal-section.svelte";
-  import { formatPrice, getRenewsLabel } from "../../helpers/price-labels";
+    import ModalSection from "../modal-section.svelte";
+    import {formatPrice, getRenewsLabel} from "../../helpers/price-labels";
+    import {getTrialsLabel} from "../../helpers/price-labels.js";
+    import { Product, PurchaseOption, SubscriptionOption } from "../../entities/offerings";
 
-  export let productDetails: any;
+    export let productDetails: Product;
+    export let purchaseOption: PurchaseOption;
+
+    const subscriptionOption: SubscriptionOption | null | undefined =
+      purchaseOption as SubscriptionOption;
+    const trial = subscriptionOption?.trial;
+    const basePrice = subscriptionOption?.base?.price;
 </script>
 
 <ModalSection>
-  <div class="rcb-pricing-info">
-    <span>{productDetails.displayName}</span>
-    <span class="rcb-product-price">
-      {productDetails.currentPrice.currency}{" "}{formatPrice(
-        productDetails.currentPrice.amount,
-        productDetails.currentPrice.currency,
-      )}
-    </span>
-    <ul class="rcb-product-details">
-      <li>Billed {getRenewsLabel(productDetails.normalPeriodDuration)}</li>
-      <li>Continues until canceled</li>
-      <li>Cancel anytime</li>
-    </ul>
-  </div>
+    <div class="rcb-pricing-info">
+        <span>{productDetails.displayName}</span>
+
+
+        <span class="rcb-product-price">
+            {#if trial?.periodDuration}
+                {getTrialsLabel(trial.periodDuration)} free trial
+            {/if}
+            {#if !trial?.periodDuration && basePrice }
+                {basePrice.currency || ''} {formatPrice(
+                    basePrice.amount,
+                    basePrice.currency,
+                )}
+            {/if}
+
+        </span>
+        {#if (trial && basePrice)}
+			<span class="rcb-product-price-after-trial">
```

Comment:
Super nit – I think there are some tab formatted lines in here while everything else uses spaces